expression/builtin_string.go

```diff
@@ -2970,9 +2970,72 @@ func evalNumDecArgsForFormat(f builtinFunc, row chunk.Row) (string, string, bool
 		d = formatMaxDecimals
 	}
 	dStr := strconv.FormatInt(d, 10)
+	xStr, err = roundFormatArgs(xStr, dStr)
+	if err != nil {
+		return "", "", isNull, err
+	}
 	return xStr, dStr, false, nil
 }
 
+func roundFormatArgs(xStr, dStr string) (string, error) {
```

Comment:
Why don't you just use `d` as its parameter.

Reply:
It mean `digit_string`.
And `dstr` can keep in line with the previous ones.

Reply:
emm.. What I mean is that why don't you use `digit int`.
Then `evalNumDecArgsForFormat` can just deliver its local variable `d` to `roundFormatArgs `.
And `roundFormatArgs` doesn't have to `Atoi` it again.